code/modules/antagonists/_common/antag_team.dm

```diff
@@ -1,24 +1,191 @@
+ */
 /datum/team
-	var/list/datum/mind/members = list()
-	var/name = "team"
-	var/member_name = "member"
-	var/list/objectives = list() //common objectives, these won't be added or removed automatically, subtypes handle this, this is here for bookkeeping purposes.
-
-/datum/team/New(starting_members)
-	. = ..()
-	if(starting_members)
-		if(islist(starting_members))
-			for(var/datum/mind/M in starting_members)
-				add_member(M)
-		else
-			add_member(starting_members)
-
-/datum/team/proc/is_solo()
-	return members.len == 1
-
-/datum/team/proc/add_member(datum/mind/new_member)
+	/// The name of the team.
+	var/name = "Generic Team Name"
+	/// A list of [minds][/datum/mind] who belong to this team.
+	var/list/members
+	/// A list of objectives which all team members share.
+	var/list/objectives
+
+/datum/team/New(list/starting_members)
+	..()
+	members = list()
+	objectives = list()
+	if(!islist(starting_members))
+		starting_members = list(starting_members)
+	for(var/datum/mind/M in starting_members)
+		add_member(M)
+	GLOB.antagonist_teams += src
+
+/datum/team/Destroy(force = FALSE, ...)
+	for(var/datum/mind/member as anything in members)
+		remove_member(member)
+	QDEL_LIST(objectives)
+	members.Cut()
+	GLOB.antagonist_teams -= src
+	return ..()
+
+/**
+ * Adds `new_member` to this team.
+ */
+/datum/team/proc/add_member(datum/mind/new_member, datum/antagonist/antag)
+	SHOULD_CALL_PARENT(TRUE)
 	members |= new_member
+	antag.objectives |= objectives
 
+/**
+ * Removes `member` from this team.
+ */
 /datum/team/proc/remove_member(datum/mind/member)
+	SHOULD_CALL_PARENT(TRUE)
+	var/datum/antagonist/A = get_antag_datum_from_member(member)
 	members -= member
+	A.objectives -= objectives
+
+/**
+ * Adds a team objective to each member's matching antag datum.
+ */
+/datum/team/proc/add_objective_to_members(datum/objective/O)
+	for(var/datum/mind/M as anything in members)
+		var/datum/antagonist/A = get_antag_datum_from_member(M)
+		A.objectives |= O
+
+/**
+ * Remove a team objective from each member's matching antag datum.
+ */
+/datum/team/proc/remove_objective_from_members(datum/objective/O)
+	for(var/datum/mind/M as anything in members)
+		var/datum/antagonist/A = get_antag_datum_from_member(M)
+		A.objectives -= O
+	objectives -= O
+	qdel(O)
+
+/**
+ * Return an antag datum from a member which is linked with this team.
+ */
+/datum/team/proc/get_antag_datum_from_member(datum/mind/member)
+	for(var/datum/antagonist/A as anything in member.antag_datums)
+		if(A.get_team() != src)
+			continue
+		return A
+
+/**
+ * Allows admins to send a message to all members of this team.
+ */
+/datum/team/proc/admin_communicate(mob/user)
+	var/message = input(user, "Enter a message to send to the team:", "Team Message") as text|null
+	if(!message)
+		return
+
+	for(var/datum/mind/M as anything in members)
+		to_chat(M.current, "<font color='#d6000b'><span class='bold'>Admin Team Message ([user.key]): </span><span class='notice'>[message]</span>")
```

Comment:
is this intended to be an `OOC` message or is it supposed to be use for IC messaging?

Reply:
An OOC message, it's basically AOOC but for this specific team only.